sdk/go/pulumi-language-go/main.go

```diff
@@ -89,43 +156,33 @@ func newLanguageHost(engineAddress, tracing string) pulumirpc.LanguageRuntimeSer
 // GetRequiredPlugins computes the complete set of anticipated plugins required by a program.
 func (host *goLanguageHost) GetRequiredPlugins(ctx context.Context,
 	req *pulumirpc.GetRequiredPluginsRequest) (*pulumirpc.GetRequiredPluginsResponse, error) {
-	return &pulumirpc.GetRequiredPluginsResponse{}, nil
-}
-
-const unableToFindProgramTemplate = "unable to find program: %s"
-
-// findProgram attempts to find the needed program in various locations on the
-// filesystem, eventually resorting to searching in $PATH.
-func findProgram(program string) (string, error) {
 
-	// look in the same directory
-	cwd, err := os.Getwd()
+	cmd, err := findProgram(req.GetProject())
 	if err != nil {
-		return "", errors.Wrap(err, "unable to get current working directory")
+		return nil, err
 	}
 
-	cwdProgram := filepath.Join(cwd, program)
-	if fileInfo, err := os.Stat(cwdProgram); !os.IsNotExist(err) && !fileInfo.Mode().IsDir() {
-		logging.V(5).Infof("program %s found in CWD", program)
-		return cwdProgram, nil
+	cmd.Env = []string{"PULUMI_PLUGINS=true"}
```

Comment:
Perhaps just a comment here to make clear this is invoking the user program just to collect plugins?  This is subtle enough that worth being clear on.